### PR TITLE
Initial carousel slider implementation

### PR DESCRIPTION
**Pull request by JsseL**
Simplistic approach for defining a Carousel model with n related slides.

### CODE REVIEW

File: cmsplugin_carousel_ai/models.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+    def has_link(self):
+        return bool(self.linked_page_id or self.url)
+
+    @property
+    def thumbnail_url(self):
+        """Return url for properly generated thumbnail"""
+        options = {
+            "size": (1920, 600),
+            "crop": True
+        }
```
Comment (JsseL) [inline cmsplugin_carousel_ai/models.py]: I'll need to come up with something more generic with this. Maybe leave the size definition for the template designer.

File: cmsplugin_carousel_ai/models.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+class Carousel(CMSPlugin):
+    """
+    Model for storing carousel meta data and configuration.
+    """
+    name = models.CharField(
+        verbose_name=_("name"),
+        max_length=160
+    )
+    interval = models.IntegerField(
```
Comment (akx) [inline cmsplugin_carousel_ai/models.py]: Might be worth it to store this is as a `FloatField` (in seconds). Seconds are more familiar to end users anyway :)

File: cmsplugin_carousel_ai/models.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        blank=True
+    )
+    linked_page = models.ForeignKey(
+        Page,
+        verbose_name=_("link to page"),
+        null=True,
+        blank=True,
+        help_text=_("Page link overrides given URL."),
+        limit_choices_to={"publisher_is_draft": False},
+        related_name="linking_slides",
```
Comment (akx) [inline cmsplugin_carousel_ai/models.py]: I'm not sure there's much point in this related_name? Or rather, what's the UC for finding pages that are linked to in slides? :)

File: cmsplugin_carousel_ai/models.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+
+class Slide(models.Model):
+    """
+    Model for storing carousel slide related data.
+    """
+    carousel = models.ForeignKey(
+        Carousel,
+        verbose_name=_("carousel"),
+        related_name="slides"
```
Comment (akx) [inline cmsplugin_carousel_ai/models.py]: Add trailing commas (everywhere), please :)

File: cmsplugin_carousel_ai/models.py (hunk 3 above)
Comment (JsseL) [reply cmsplugin_carousel_ai/models.py]: Just a habit to give reasonable related_name for future needs.

Comment (JsseL) [reply cmsplugin_carousel_ai/models.py]: Removed it.

**Comment by JsseL** [comment]
Further comments?

File: cmsplugin_carousel_ai/models.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
+# -*- coding: utf-8 -*-
+from django.db import models
+from django.utils.translation import ugettext_lazy as _
+
+from cms.models import CMSPlugin, Page, force_text
+from easy_thumbnails.files import get_thumbnailer
```
Comment (akx) [inline cmsplugin_carousel_ai/models.py]: Looks like an unused import. Have you run the usual code style tools on the source code?

Comment (JsseL) [reply cmsplugin_carousel_ai/models.py]: See the new commit.

**Pull request merged**